src/ImmutablePropTypes.js

```diff
@@ -171,23 +172,23 @@ function createRecordOfTypeChecker(recordKeys) {
 }
 
 // there is some irony in the fact that shapeTypes is a standard hash and not an immutable collection
-function createShapeTypeChecker(shapeTypes) {
+function createShapeTypeChecker(shapeTypes, immutableClassName = 'Iterable', immutableClassTypeValidator = Immutable.Iterable.isIterable) {
   function validate(props, propName, componentName, location) {
     var propValue = props[propName];
     var propType = getPropType(propValue);
-    if (!Immutable.Iterable.isIterable(propValue)) {
+    if (!immutableClassTypeValidator(propValue)) {
       var locationName = location;
       return new Error(
         `Invalid ${locationName} \`${propName}\` of type \`${propType}\` ` +
-        `supplied to \`${componentName}\`, expected an Immutable.js Iterable.`
+        `supplied to \`${componentName}\`, expected an Immutable.js ${immutableClassName}.`
       );
     }
+    var mutablePropValue = propValue.toJS();
```

Comment:
This morning I realized I should probably convert this back to `.toObject`. I hadn't realized before today that it only shallowly converts. Which allows you to continue to propType check deeply for Immutable objects.

`Immutable.fromJS({a: [1, 2]}).toObject() => { a: List [ 1, 2 ] }`
`Immutable.fromJS({a: [1, 2]}).toJS() => { a: [ 1, 2 ] }`